lib/connextclient/ConnextClient.ts

```diff
@@ -268,6 +268,28 @@ class ConnextClient extends SwapClient {
     return this._totalOutboundAmount.get(currency) || 0;
   }
 
+  /**
+   * Checks whether we have a pending collateral request for the currency and,
+   * if one doesn't exist, starts a new request for the specified amount. Then
+   * calls channelBalance to refresh the inbound capacity for the currency.
+   */
+  private requestCollateralInBackground = (currency: string, units: number) => {
+    // first check whether we already have a pending collateral request for this currency
+    // if not start a new request, and when it completes call channelBalance to refresh our inbound capacity
+    const requestCollateralPromise = this.requestCollateralPromises.get(currency) ?? this.sendRequest('/request-collateral', 'POST', {
+      assetId: this.tokenAddresses.get(currency),
+      amount: units.toLocaleString('fullwide', { useGrouping: false }),
+    }).then(() => {
+      this.requestCollateralPromises.delete(currency);
```

Comment:
nit: can we move this to `.finally` block? That way we can also remove it from the `catch` block.

Reply:
I had it like that originally, but I changed it back to this because the `then` block is also doing and returning the `channelBalance` call but I want to delete the promise immediately after the collateral request completes and not have to wait for `channelBalance`. The only way I could think to do that is with two separate `catch` blocks and a `finally` block or like this, and I went with the second option. Let me know if that makes sense or if you have a better idea.